pkg/engine/health/health.go

```diff
@@ -106,7 +108,7 @@ func IsHealthy(obj runtime.Object) error {
 		return fmt.Errorf("instance %s/%s active plan is in state %v", obj.Namespace, obj.Name, obj.Spec.PlanExecution.Status)
 
 	case *corev1.Pod:
-		if obj.Status.Phase == corev1.PodRunning {
+		if obj.Status.Phase == corev1.PodRunning && podutils.IsPodReady(obj) {
```

Comment:
Do we still need both conditions to be true? Isn't `podutils.IsPodReady(obj)` enough?

Reply:
I did not have that in the first version but in the end I ended up doing exactly what StatefulSet is doing to be on the safe side https://github.com/kubernetes/kubernetes/blob/60044a8accfe8fddeb452c4adca4fefbbff64967/pkg/controller/statefulset/stateful_set_utils.go#L204